internal/cli/atlas/search/nodes/create.go

```diff
@@ -0,0 +1,127 @@
+	"github.com/mongodb/mongodb-atlas-cli/internal/flag"
+	"github.com/mongodb/mongodb-atlas-cli/internal/store"
+	"github.com/mongodb/mongodb-atlas-cli/internal/usage"
+	"github.com/spf13/afero"
+	"github.com/spf13/cobra"
+)
+
+type CreateOpts struct {
+	cli.GlobalOpts
+	cli.WatchOpts
+	clusterName string
+	filename    string
+	fs          afero.Fs
+	store       store.SearchNodesCreator
+}
+
+const (
+	defaultWatchTimeoutSeconds = 600 // 10 minutes
+	stateIdle                  = "IDLE"
+)
+
+func (opts *CreateOpts) initStore(ctx context.Context) func() error {
+	return func() error {
+		var err error
+		opts.store, err = store.New(store.AuthenticatedPreset(config.Default()), store.WithContext(ctx))
+		return err
+	}
+}
+
+var createTemplate = "Created new search nodes.\n"
+
+func (opts *CreateOpts) Run() error {
+	spec, err := loadAPISearchDeploymentSpec(opts.fs, opts.filename)
+	if err != nil {
+		return err
+	}
+
+	r, err := opts.store.CreateSearchNodes(opts.ConfigProjectID(), opts.clusterName, spec)
+	if err != nil {
+		return err
+	}
+
+	if opts.EnableWatch {
+		if err := opts.Watch(opts.watcher); err != nil {
+			return err
+		}
+	}
+
+	return opts.Print(r)
+}
+
+func (opts *CreateOpts) watcher() (bool, error) {
+	res, err := opts.store.SearchNodes(opts.ConfigProjectID(), opts.clusterName)
+	if err != nil {
+		return false, err
+	}
+	return *res.StateName == stateIdle, nil
+}
+
+// atlas clusters search nodes create [--clusterName clusterName] [--file filePath].
+func CreateBuilder() *cobra.Command {
+	opts := &CreateOpts{}
+	opts.fs = afero.NewOsFs()
+
+	cmd := &cobra.Command{
+		Use:   "create",
+		Short: "Create a search node for a cluster.",
+		Long:  fmt.Sprintf(usage.RequiredRole, "Organization Owner or Project Owner"),
+		Args:  require.NoArgs,
+		Annotations: map[string]string{
+			"output": createTemplate,
+		},
+		Example: fmt.Sprintf(`  # Create a search node for the cluster named myCluster using a JSON node spec configuration file named spec.json:
+  %s clusters search nodes create --clusterName myCluster --file spec.json --output json`, cli.ExampleAtlasEntryPoint()),
+		PreRunE: func(cmd *cobra.Command, _ []string) error {
+			return opts.PreRunE(
+				opts.ValidateProjectID,
+				opts.initStore(cmd.Context()),
+				opts.InitOutput(cmd.OutOrStdout(), createTemplate),
+			)
+		},
+		RunE: func(_ *cobra.Command, _ []string) error {
+			return opts.Run()
+		},
+	}
+
+	// Command specific flags
+	cmd.Flags().StringVar(&opts.clusterName, flag.ClusterName, "", usage.ClusterName)
+	_ = cmd.MarkFlagRequired(flag.ClusterName)
+
+	cmd.Flags().StringVarP(&opts.filename, flag.File, flag.FileShort, "", usage.SearchNodesFilename)
+	_ = cmd.MarkFlagFilename(flag.File)
+	_ = cmd.MarkFlagRequired(flag.File)
+
+	cmd.Flags().BoolVarP(&opts.EnableWatch, flag.EnableWatch, flag.EnableWatchShort, false, usage.EnableWatchDefault)
+	cmd.Flags().UintVar(&opts.Timeout, flag.WatchTimeout, defaultWatchTimeoutSeconds, usage.WatchTimeout)
```

Comment:
[nit] all watchers have default timeout=0, which means wait indefinitely. But users can configure their own timeout (e.g. --watchTimeout 600). For consistency let's keep the same behavior here too.

```suggestion
	cmd.Flags().UintVar(&opts.Timeout, flag.WatchTimeout, 0, usage.WatchTimeout)
```

Reply:
updated